app/serializers/v1/fee_serializer.rb

```diff
@@ -52,6 +52,18 @@ def serialize
     private
 
     def date_boundaries
+      if model.charge? && !model.pay_in_advance? && model.charge.pay_in_advance?
```

Comment:
I might be misunderstanding this one, but since we are in the `FeeSerializer`, I think that model is a fee, and I'm not sure it as a `pay_in_advance` attribute. Should it be `model.subscription.pay_in_advance?`

Reply:
Seems strange but actually it is good :) Fee can also be pay in advance for the event pay in advance invoice (bot the one at the end of the billing period). These boundaries should be only recalculated for charges that are pay in advance...